libs/iota-browser/src/helpers/channel-provider.ts

```diff
@@ -178,27 +189,44 @@ export class ChannelProvider {
             )
             try {
               const event = JSON.parse(raw_data)
-              if (
-                event.eventType === 'signedRequest' &&
-                correlationId === event.correlationId
-              ) {
-                // TODO handle Zod errors gracefully
-                const signedRequest = SignedRequestEventSchema.parse(event)
-                const claims = jose.decodeJwt(signedRequest.data.jwt)
-                // TODO Zod validate JWT
-                if (!claims.client_id && typeof claims.client_id === 'string') {
-                  reject(new Error('Unexpected request claims received'))
+              if (correlationId !== event.correlationId) {
+                return
+              }
+              if (event.eventType === EventTypes.SignedRequest) {
+                let signedRequest: SignedRequestEvent,
+                  signedRequestJWT: SignedRequestJWT
+                try {
+                  signedRequest = SignedRequestEventSchema.parse(event)
+                } catch (e) {
+                  throw Error(
+                    getUnexpectedErrorMessage(ErrorCode.SIGNED_REQUEST_EVENT),
+                  )
+                }
+                try {
+                  const claims = jose.decodeJwt(signedRequest.data.jwt)
+                  signedRequestJWT = SignedRequestJWTSchema.parse(claims)
+                } catch (e) {
+                  throw Error(
+                    getUnexpectedErrorMessage(ErrorCode.SIGNED_REQUEST_JWT),
+                  )
```

Comment:
if untilately it's supposed to throw an exception, we could maybe abstract this away into a dedicated function. wdyt ?